citesphere/src/main/java/edu/asu/diging/citesphere/core/service/IAuthorityService.java

```diff
@@ -41,12 +43,60 @@ AuthoritySearchResult searchAuthorityEntries(IUser user, String firstName, Strin
+    /**
+     * Finds the authorities whose name matches the given first and last name and
+     * which are created or imported by the given user
+     * @param user      User who created or imported the authority
+     * @param firstName First name by which the authorities are to be searched
+     * @param lastName  Last name by which the authorities are to be searched
+     * @param page      Requested page number
+     * @param pageSize  Size of the requested page
+     * @return Requested page of the found authorities
+     */
+    Page<IAuthorityEntry> findByFirstNameAndLastName(IUser user, String firstName, String lastName, int page, int pageSize);
+    
+    /**
+     * Finds the authorities whose name matches the given last name but does not
+     * match the given first name and which are created or imported by the given user
+     * @param user      User who created or imported the authority
+     * @param firstName First name by which the authorities are to be excluded
+     * @param lastName  Last name by which the authorities are to be searched
+     * @param page      Requested page number
+     * @param pageSize  Size of the requested page
+     * @return Requested page of the found authorities
+     */
+    Page<IAuthorityEntry> findByLastNameAndExcludingFirstName(IUser user, String firstName, String lastName, int page, int pageSize);
+    
+    int getTotalUserAuthoritiesPages(IUser user, String firstName, String lastName, int pageSize);
+    
+    /**
+     * Finds the authorities who belong to the given group and whose name matches
+     * the given first and last name
+     * @param groupId   Group Id to which the authority should belong to
+     * @param firstName First name by which the authorities are to be searched
+     * @param lastName  Last name by which the authorities are to be searched
+     * @param page      Requested page number
+     * @param pageSize  Size of the requested page
+     * @return Requested page of the found authorities
+     */
+    Page<IAuthorityEntry> findByGroupAndFirstNameAndLastName(Long groupId, String firstName, String lastName, int page, int pageSize);
+    
+    /**
+     * Finds the authorities who belong to the given group and whose name matches
+     * the given last name but does not match the given first name
+     * @param groupId   Group Id to which the authority should belong to
+     * @param firstName First name by which the authorities are to be excluded
+     * @param lastName  Last name by which the authorities are to be searched
+     * @param page      Requested page number
+     * @param pageSize  Size of the requested page
+     * @return Requested page of the found authorities
+     */
+    Page<IAuthorityEntry> findByGroupAndLastNameAndExcludingFirstName(Long groupId, String firstName, String lastName, int page, int pageSize);
+    
+    int getTotalGroupAuthoritiesPages(Long groupId, String firstName, String lastName, int pageSize);
```

Comment:
get total methods need javadoc. It needs to explicitly. state that the total is only searching for first or last name. This is not intuitive or recognizable from the method name.